### PR TITLE
Hail query, no vep

### PR DESCRIPTION
**Pull request by MattWellie**
Replaces https://github.com/populationgenomics/automated-interpretation-pipeline/pull/5, but without any Annotation phase

It was discovered that the annotation provided by VEP is insufficient, and recreating the exact annotation provided by the Seqr pipeline will require some thought (increases dependencies on external data sources). 

Instead of solving that issue now, this PR is a route to getting the end-to-end process merged into this repository with the known limitation that the annotation is not provided, rather than committing code which is known to be faulty.

# Fixes #

  - https://github.com/populationgenomics/automated-interpretation-pipeline/issues/4
  - does the complete opposite of https://github.com/populationgenomics/automated-interpretation-pipeline/issues/9 - further increases reliance on seqr pipeline
  - https://github.com/populationgenomics/automated-interpretation-pipeline/issues/6 ... sort of, but definitely cheating

## Proposed Changes ##

  - Replaces https://github.com/populationgenomics/automated-interpretation-pipeline/pull/5, but without any Annotation phase

## Checklist ##

- [x] Related Issue created
- [x] Tests covering new change
- [x] Linting checks pass

### CODE REVIEW

File: design_docs/Configuration.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+## `moi_tests`
+
+This section contains relates to the application of Mode Of Inheritance tests. These are not set based on the prevalence
+any individual disease in any specific population. Feedback from clinical teams will be used to further refine these
+settings
+
+* gnomad_dominant - the max gnomad population AF permitted for Mono-allelic disease
+* gnomad_max_homs_dominant - max number of Homs in Gnomad for Mono-allelic disease
+* gnomad_max_homs_recessive - max number of Homs in Gnomad for Bi-allelic disease
+* gnomad_max_homs_recessive - max number of variants in Gnomad for Mono-allelic disease
```
Comment (lgruen) [inline design_docs/Configuration.md:16]:
```suggestion
* gnomad_max_ac_dominant - max number of variants in Gnomad for Mono-allelic disease
```

File: design_docs/Configuration.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+logic, so that the fields don't need to be enumerated in config, which feels brittle.
+
+## `filter`
+
+This section relates to the variant filtration and categorisation process, currently executed in Hail query. These
+default settings are primarily based on a clinical-team provided specification, and can be replaced for a bespoke run.
+
+* af_semi_rare - a minimum AF threshold to be applied to all variants
+* min_samples_to_ac_filter - minimum number of samples in the joint-call in order to apply common-in-cohort filter
+* ac_threshold - where the joint-call filter is applied, remove all variants more common than this in the callset
```
Comment (lgruen) [inline design_docs/Configuration.md:43]: "more common than this" is a bit unclear -- what does this threshold represent? Is it a fraction (in [0,1]) of samples that need to have at least one copy of the variant?

File: design_docs/Configuration.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+* ref_genome - ref_genome
+* useless_csq - list of VEP consequences filtered out prior to classification steps
+* critical_csq - list of VEP consequences treated as high priority/high impact
+* in_silico - separate sub-group, thresholds to apply for each *in silico* tool. These parameters are taken from [work by *Pejaver et al.*](https://www.biorxiv.org/content/10.1101/2022.03.17.484479v1)
+
+## `output`
+
+This section relates to the rubbish HTML document created as output. To be discontinued ASAP
```
Comment (lgruen) [inline design_docs/Configuration.md:51]: 🤣

File: design_docs/Configuration.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+* seqr_lookup - location of a cache file containing the CPG_ID: SEQR_family_ID mapping
+* colours - Hex codes for text colours in the final HTML document
```
Comment (lgruen) [inline design_docs/Configuration.md:54]: Do we want to use British spelling for variable names? 🇬🇧

File: reanalysis/wrapper.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+from cpg_utils.hail import init_batch, output_path
+
+from query_panelapp import main as panelapp_main
+from hail_filter_and_categorise import main as category_main
+
+
+DEFAULT_IMAGE = os.getenv('CPG_DRIVER_IMAGE')
+assert DEFAULT_IMAGE
+
```
Comment (lgruen) [inline reanalysis/wrapper.py:32]:
```suggestion
```

File: reanalysis/wrapper.py [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+# static paths to write outputs
+PANELAPP_JSON_OUT = output_path('panelapp_137_data.json')
+HAIL_VCF_OUT = output_path('hail_categorised.vcf.bgz')
+COMP_HET_JSON = output_path('hail_comp_het.json')
+CONFIG_OUT = output_path('config_used.json')
+REHEADERED_OUT = output_path('hail_categories_reheadered.vcf.bgz')
+MT_TMP = output_path('tmp_hail_table.mt', category='tmp')
+
+# location of the Slivar Docker image
```
Comment (lgruen) [inline reanalysis/wrapper.py:41]:
```suggestion
```

File: reanalysis/wrapper.py [hunk 7]
```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python3
+
+
+"""
+wrapper for reanalysis process
```
Comment (lgruen) [inline reanalysis/wrapper.py:5]: Let's maybe not call this wrapper, but it's really the point where the pipeline is constructed, right?

I think we might want to reserve the wrapper for a non-CPG replacement of the analysis-runner, which sets up the right environment variables to launch this script.

File: reanalysis/wrapper.py [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+# static paths to write outputs
+PANELAPP_JSON_OUT = output_path('panelapp_137_data.json')
+HAIL_VCF_OUT = output_path('hail_categorised.vcf.bgz')
+COMP_HET_JSON = output_path('hail_comp_het.json')
+CONFIG_OUT = output_path('config_used.json')
+REHEADERED_OUT = output_path('hail_categories_reheadered.vcf.bgz')
+MT_TMP = output_path('tmp_hail_table.mt', category='tmp')
+
+# location of the Slivar Docker image
+AR_REPO = 'australia-southeast1-docker.pkg.dev/cpg-common/images'
```
Comment (lgruen) [inline reanalysis/wrapper.py:42]: We should probably fetch that from an environment variable, too. I can prepare that for the analysis-runner.

File: reanalysis/wrapper.py [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+    # :param ped_file:
+    """
+
+    logging.info('Starting the reanalysis batch')
+
+    config_dict = read_json_dict_from_path(config_json)
+
+    service_backend = hb.ServiceBackend(
+        billing_project=os.getenv('HAIL_BILLING_PROJECT'),
+        bucket=os.getenv('HAIL_BUCKET'),
```
Comment (lgruen) [inline reanalysis/wrapper.py:181]: This is deprecated. You can import `remote_tmpdir` from `cpg_utils` and instead use:
```suggestion
        remote_tmpdir=remote_tmpdir()
```

**Review by lgruen — APPROVED** [review]
This looks good to me -- just added a few nit comments. It would be great to get @vladsaveliev's feedback for this too, but he'll be on leave for the rest of the week. So that shouldn't block you, especially if you're planning on refactoring this to use Vlad's modular pipeline framework, at which point you'll probably get this reviewed again anyhow :).

File: design_docs/Configuration.md (hunk 4 above)
Comment (MattWellie) [reply design_docs/Configuration.md:54]: I do but I suspect I'm in the minority... though I'm actually not sure whether Australian spelling trends towards 🇺🇸  or 🏴󠁧󠁢󠁥󠁮󠁧󠁿

Comment (MattWellie) [reply design_docs/Configuration.md:54]: crucially in code it will be American, so happy to change

**Pull request merged**